services/agents/registry.go

```diff
@@ -268,8 +268,36 @@ func (r *Registry) Run(stream agentpb.Agent_ConnectServer) error {
 					Payload: new(agentpb.ActionResultResponse),
 				})
 
+			case *agentpb.JobResult:
+				if e := r.db.InTransaction(func(t *reform.TX) error {
```

Comment:
maybe extract this to new function?
case inside another case, seems too complex

Reply:
Done